pkg/minishift/setup/hypervisor/virtualization_darwin.go

```diff
@@ -17,15 +17,80 @@ limitations under the License.
 package hypervisor
 
 import (
+	"errors"
 	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"os/user"
+
+	"github.com/minishift/minishift/pkg/util/progressdots"
+)
+
+const (
+	driverBinaryDir   = "/usr/local/bin"
+	driverBinaryPath  = driverBinaryDir + "/docker-machine-driver-xhyve"
+	driverDownloadUrl = "https://github.com/machine-drivers/docker-machine-driver-xhyve/releases/download/v0.3.3/docker-machine-driver-xhyve"
```

Comment:
@anjannath The only concern I have and discussed (with @gbraad ) is this pre-compiled binary might not work across all the MacOS versions :( I don't know what would be the best solution. Might be we host this on somewhere according to mac version and then fetch it?

Reply:
@praveenkumar My thoughts as well, although i tried this binary in macOS mojave and it worked out of the box. But we should recompile this for specific versions of macOS and do the release in _machine-drivers_ org.

Then we can fetch the respective driver based on version of macOS.

Reply:
@anjannath Lets create a follow up issue to handle this issue. I think this an important issue else we will end up getting issues for `minishift setup`.